static/js/protein/data_processing_utils.ts

```diff
@@ -592,3 +530,328 @@ export function getProteinDescription(data: GraphNodes): string {
     return proteinDescription;
   }
 }
+
+/**
+ * Given array, key-maker function, and value-maker function, construct object from array with string-valued keys.
+ * This is essentially a TS implementation of a python dictionary comprehension.
+ *
+ * Unfortunately generic index signature parameters aren't supported by TS yet, so we enforce that keys are strings
+ * (since they're automatically converted to strings anyway).
+ *
+ * References:
+ *  - https://linuxtut.com/en/87ab3f313a75b547c278/
+ *  - https://stackoverflow.com/questions/13315131/enforcing-the-type-of-the-indexed-members-of-a-typescript-object
+ *
+ */
+export function objectFromArray<T, V>(
+  arr: T[],
+  keyFunc: (arrElt: T) => string,
+  valFunc: (arrElt: T) => V
+): { [key: string]: V } {
+  return arr.reduce(
+    (obj, elt) => Object.assign(obj, { [keyFunc(elt)]: valFunc(elt) }),
+    {}
+  );
+}
+
+/**
+ * Given an (unpacked) array of arrays, return its transpose (after trimming arrays to length of shortest one).
+ * This is essentially a TS implementation of python's built-in zip function.
+ *
+ * Adapted from implementation by Matt Mancini
+ *
+ * Thanks also to Carlos Israel Ortiz Garcia and Jack Davison for the below references:
+ *  - https://ncjamieson.com/typing-zipwith/#typing-zip
+ *  - https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-0.html#variadic-tuple-types
+ */
+// Recursive elementwise type inference on array of arrays
+type Zip<T> = T extends [Array<infer U>, ...infer Z] ? [U, ...Zip<Z>] : [];
+export function zip<T extends unknown[][]>(...arrays: T): Array<Zip<T>> {
```

Comment:
i don't think you need to create your own zip method since lodash has one: https://lodash.com/docs/4.17.15#zip

Reply:
Oh oops thanks didn't know! Will just delete this and use lodash's then unless I find an example for which this does better type inference than lodash.